bindings/matrix-sdk-ffi/src/room.rs

```diff
@@ -902,6 +902,91 @@ impl Room {
             Ok(Arc::new(RoomMessageEventContent::new(msgtype)))
         })
     }
+
+    pub async fn can_user_redact(&self, user_id: String) -> Result<bool, ClientError> {
+        let room = match &self.inner {
+            SdkRoom::Joined(j) => j.clone(),
+            _ => return Err(anyhow!("Can't check permissions for non joined rooms").into()),
+        };
```

Comment:
nit: the clone is probably not required here, plus it's more idiomatic to let anyhow return the error instead of producing one, wrap it in `Err` and return it:

```suggestion
        let SdkRoom::Joined(room) = &self.inner else { anyhow::bail!("Can't check permissions for non joined rooms"); };
```

Reply:
`bail!` doesn't work because we're returning `ClientError`, not `anyhow::Error`.